packages/cli-repl/src/logger.ts

```diff
@@ -1,28 +1,92 @@
 import redactInfo from 'mongodb-redact';
+import Analytics from 'analytics-node';
 import redactPwd from './redact-pwd';
 import { ObjectId } from 'bson';
 import pino from 'pino';
 import clr from './clr';
 import path from 'path';
 import os from 'os';
 
+interface ApiCallArguments {
+  pipeline?: any[];
+  query?: object;
+  options?: object;
+  filter?: object; 
+}
+
+interface ApiCall {
+  method?: string; 
+  class?: string;
+  db?: any;
+  coll?: any;
+  arguments?: ApiCallArguments;
+}
+
 function logger(bus: any, logDir: string) {
   const sessionID = new ObjectId(Date.now());
   const logDest = path.join(logDir, `${sessionID}_log`);
   const log = pino({ name: 'monogsh' }, pino.destination(logDest));
   console.log(`Current sessionID: ${sessionID}`);
+  let userId;
+  let telemetry;
+
+  let analytics = new NoopAnalytics();
```

Comment:
use a noop just in case something goes wrong with the definition below. That way the following `analytics` calls don't fail and pollute user with errors.